src/main/java/de/unistuttgart/iste/gits/quiz_service/service/QuizService.java

```diff
@@ -270,4 +277,84 @@ private QuestionEntity getQuestionInQuizByNumber(QuizEntity quizEntity, int numb
                                 "Question with number {0} not found in quiz with id {1}.",
                                 number, quizEntity.getAssessmentId())));
     }
+
+    /**
+     * removes all Quizzes when linked Content gets deleted
+     *
+     * @param dto event object containing changes to content
+     */
+    public void removeContentIds(ContentChangeEvent dto) {
+
+        // validate event message
+        try {
+            checkCompletenessOfDto(dto);
+        } catch (NullPointerException e) {
+            log.error(e.getMessage());
+            return;
+        }
+        // only consider DELETE Operations
+        if (!dto.getOperation().equals(CrudOperation.DELETE) || dto.getContentIds().isEmpty()) {
+            return;
+        }
+
+        // find all quizzes
+        List<QuizEntity> quizEntities = quizRepository.findQuizEntitiesByAssessmentIdIn(dto.getContentIds());
```

Comment:
assessmentId is the primary key of the quiz.  It is always at most one entity. Using a list here is just confusing in my oppinion

Reply:
I just noticed that too. I will refactor this